src/core/settings/qgssettingsregistrycore.cpp

```diff
@@ -132,14 +126,240 @@ QgsSettingsRegistryCore::QgsSettingsRegistryCore()
   addSettingsEntry( &QgsGpsConnection::settingsGpsTimeStampTimeZone );
   addSettingsEntry( &QgsGpsConnection::settingsGpsTimeStampOffsetFromUtc );
 
-  addSettingsEntry( &QgsBabelFormatRegistry::settingsBabelDeviceList );
-  addSettingsEntryGroup( &QgsBabelFormatRegistry::settingsBabelDeviceGroup );
-
   addSettingsEntry( &QgsGpsLogger::settingsGpsStoreAttributeInMValues );
   addSettingsEntry( &QgsGpsLogger::settingsGpsMValueComponent );
+
+  migrateOldSettings();
 }
 
 QgsSettingsRegistryCore::~QgsSettingsRegistryCore()
 {
+  backwardCompatibility();
+}
+
+void QgsSettingsRegistryCore::migrateOldSettings()
+{
+  // the logic is to copy value from old key to new keys and delete the old ones
+  // for backward compatibility, old keys are recreated when the registry gets deleted
```

Comment:
```suggestion
  // copy values from old keys (QGIS <3.30) to new keys (QGIS >=3.30) and delete the old ones
  // for backward compatibility, old keys are recreated when the registry gets deleted
```

Reply:
I will remove the mention of version, since potentially not everything will be done for 3.30
but for each block, it is specified in a comment